### PR TITLE
#967 Fix table button spacing and styling

### PR DESCRIPTION
**Pull request by matthoward366**
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: canvas_modules/common-canvas/src/common-properties/panels/sub-panel/cell.jsx [hunk 1]
```diff
@@ -67,26 +67,22 @@ export default class SubPanelCell extends React.Component {
...
-							kind="secondary"
-							onClick={this.showSubPanel}
-							disabled={disabled}
-							aria-label={subPanelToolTip}
-						>
-							{innerObject}
-						</Button>
-					</Tooltip>
-				</div>
+				<Tooltip
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/common-properties/panels/sub-panel/cell.jsx:70]: This is updating the edit button in the table row.  This uses a standard carbon icon button now.

File: canvas_modules/common-canvas/__tests__/common-properties/components/table-buttons-test.js [hunk 2]
```diff
@@ -134,13 +134,15 @@ describe("Table buttons renders correctly", () => {
 				"enable": false,
 				"label": "Label btn",
 				"incLabelWithIcon": "before",
+				"kind": "ghost",
```
Comment (matthoward366) [inline canvas_modules/common-canvas/__tests__/common-properties/components/table-buttons-test.js:137]: For buttons with labels on tables we should use ghost buttons

File: canvas_modules/common-canvas/src/common-properties/components/field-picker/field-picker.jsx [hunk 3]
```diff
@@ -140,43 +140,40 @@ export default class FieldPicker extends React.Component {
 		for (let i = 0; i < fields.length; i++) {
 			const field = fields[i];
 			const columns = [];
-			let fieldContent = (
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/common-properties/components/field-picker/field-picker.jsx:143]: This is fixing a couple issues.
1. Spacing where the icon would still take up space even when there wasn't an icon
2. Add a tooltip on all items that have an overflow
3. Only set the icon if type isn't the dmIcon value

File: canvas_modules/common-canvas/src/common-properties/components/field-picker/field-picker.scss [hunk 4]
```diff
@@ -18,33 +18,29 @@
 	height: 100%;
 }
 
-.properties-wf-content .properties-fp-table .properties-ft-container-absolute {
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/common-properties/components/field-picker/field-picker.scss:21]: We use 100% instead

File: canvas_modules/common-canvas/src/common-properties/controls/abstract-table.jsx [hunk 5]
```diff
@@ -773,15 +770,13 @@ export default class AbstractTable extends React.Component {
...
 		const subItemButton = this.props.buildUIItem(rowIndex, this.props.control.childItem, propertyId, this.indexOfColumn);
-		const settingsIcon = <Icon type={CARBON_ICONS.SETTINGS} />;
 		// Hack to decompose the button into our own in-table link
 		const subCell = (
 			<div className="properties-table-subcell">
 				<SubPanelCell
 					label={subItemButton.props.label}
 					title={subItemButton.props.title}
 					panel={subItemButton.props.panel}
-					buttonIcon={settingsIcon}
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/common-properties/controls/abstract-table.jsx:784]: Button icon now in the SubPanelCell class

File: canvas_modules/common-canvas/src/common-properties/components/flexible-table/flexible-table.scss [hunk 6]
```diff
@@ -39,8 +39,16 @@ $row-left-padding: 4px;
...
+	height: calc(100% - $spacing-07); // adjust height for table header
+}
+
+.properties-ft-control-container {
+	height: 100%;
+}
+
 .properties-ft-container-wrapper {
-	height: 400px;
+	// height: 400px;
```
Comment (nmgokhale) [inline canvas_modules/common-canvas/src/common-properties/components/flexible-table/flexible-table.scss:51]: This line can be removed

**Review by nmgokhale — APPROVED** [review]
Looks good.

**Pull request merged**